test/unit/org/apache/cassandra/db/guardrails/GuardrailsTest.java

```diff
@@ -0,0 +1,300 @@
+        assertAborts(() -> disallowed.guard(set(2), action, userClientState), message);
+        assertValid(() -> disallowed.guard(set(2), action, systemClientState));
+        assertValid(() -> disallowed.guard(set(2), action, superClientState));
+
+        Set<Integer> allowedValues = set(1);
+        assertValid(() -> disallowed.guard(allowedValues, action, null));
+        assertValid(() -> disallowed.guard(allowedValues, action, userClientState));
+        assertValid(() -> disallowed.guard(allowedValues, action, systemClientState));
+        assertValid(() -> disallowed.guard(allowedValues, action, superClientState));
+
+        Set<Integer> disallowedValues = set(2);
+        message = "Provided values [2] are not allowed for integer (disallowed values are: [2])";
+        assertAborts(() -> disallowed.guard(disallowedValues, action, null), message);
+        assertAborts(() -> disallowed.guard(disallowedValues, action, userClientState), message);
+        assertValid(() -> disallowed.guard(disallowedValues, action, systemClientState));
+        assertValid(() -> disallowed.guard(disallowedValues, action, superClientState));
+    }
+
+    @Test
+    public void testIgnoredValues() throws Throwable
+    {
+        // Using a sorted set below to ensure the order in the error message checked below are not random
+        Values<Integer> ignored = new Values<>(state -> new ValuesConfig(sortedSet(4, 6, 20), Collections.emptySet()),
+                                               "integer");
+
+        Set<Integer> triggeredOn = set();
+        assertValid(() -> ignored.guard(set(3), triggeredOn::add, userClientState));
+        assertEquals(set(), triggeredOn);
+
+        assertWarns(() -> ignored.guard(set(4), triggeredOn::add, userClientState),
+                    "Ignoring provided values [4] as they are not supported for integer (ignored values are: [4, 6, 20])");
+        assertEquals(set(4), triggeredOn);
+        triggeredOn.clear();
+
+        assertWarns(() -> ignored.guard(set(4, 6), triggeredOn::add, null),
+                    "Ignoring provided values [4, 6] as they are not supported for integer (ignored values are: [4, 6, 20])");
+        assertEquals(set(4, 6), triggeredOn);
+        triggeredOn.clear();
+
+        assertWarns(() -> ignored.guard(set(4, 5, 6, 7), triggeredOn::add, null),
+                    "Ignoring provided values [4, 6] as they are not supported for integer (ignored values are: [4, 6, 20])");
+        assertEquals(set(4, 6), triggeredOn);
+        triggeredOn.clear();
+    }
+
+    private static Set<Integer> set(Integer value)
+    {
+        return Collections.singleton(value);
+    }
+
+    private static Set<Integer> set(Integer... values)
+    {
+        return new HashSet<>(Arrays.asList(values));
+    }
+
+    @SafeVarargs
+    private static <T> Set<T> sortedSet(T... values)
+    {
+        return new LinkedHashSet<>(Arrays.asList(values));
+    }
```

Comment:
nit; this is not a `java.util.SortedSet` but instead an insert-ordered set... so is kinda confusing.  Doesn't impact the test, just jumped out to me.